lib/socket.js

```diff
@@ -84,7 +85,7 @@ class Transport extends EventEmitter {
 
 const socketFactory = (connect) => {
   const resultConnectFn = (...options) => {
-    const callback = common.extractCallback(options);
+    const callback = common.cbExtract(options);
```

Comment:
`extractCallback` and `cbExtract` are not equivalent, the latter has a lot of checks that are completely unnecessary here. I think it's better to place `extractCallback` in common and rename `cbExtract` to something that will represent its purpose, like `cbWrapExtract`.